controllers/bgppeer_controller.go

```diff
@@ -0,0 +1,165 @@
+	Namespace string
+}
+
+var BGPPeerManifestPath = "./bindata/configuration/bgp-peer"
+
+//+kubebuilder:rbac:groups=metallb.io,resources=bgppeers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=metallb.io,resources=bgppeers/status,verbs=get;update;patch
+
+func (r *BGPPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Info(fmt.Sprintf("Starting BGP Peer reconcile loop for %v", req.NamespacedName))
+
+	instance := &metallbv1alpha1.BGPPeer{}
+	defer r.Log.Info(fmt.Sprintf("Finish BGP Peer reconcile loop for %v", req.NamespacedName))
+
+	if err := r.Get(ctx, req.NamespacedName, instance); err != nil {
+		if errors.IsNotFound(err) {
+			err = r.syncMetalLBBGPPeers(req)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	err := r.syncMetalLBBGPPeer(instance)
+	if err != nil {
+		r.Log.Info(fmt.Sprintf("sync MetalLB BGP Peer failed %s", err))
+		return ctrl.Result{RequeueAfter: RetryPeriod}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *BGPPeerReconciler) renderObject(instance *metallbv1alpha1.BGPPeer) ([]*unstructured.Unstructured, error) {
+	data := render.MakeRenderData()
+	data.Data["Address"] = instance.Spec.Address
+	data.Data["ASN"] = instance.Spec.ASN
+	data.Data["MyASN"] = instance.Spec.MyASN
+	data.Data["Port"] = instance.Spec.Port
+	data.Data["HoldTime"] = instance.Spec.HoldTime
+	data.Data["RouterID"] = instance.Spec.RouterID
+	data.Data["Password"] = instance.Spec.Password
+	data.Data["NodeSelectors"] = instance.Spec.NodeSelectors
+	data.Data["NameSpace"] = r.Namespace
+	data.Data["ConfigMapName"] = MetalLBConfigMapName
+	data.Data["SrcAddress"] = instance.Spec.SrcAddress
+
+	objs, err := render.RenderDir(BGPPeerManifestPath, &data)
+	if err != nil {
+		return nil, fmt.Errorf("Fail to render bgp peer manifest err %v", err)
+	}
+
+	if len(objs) > 1 {
+		return nil, fmt.Errorf("Fail to render we are expecting only one object and get %d", len(objs))
+	}
+
+	return objs, err
+}
+
+func (r *BGPPeerReconciler) syncMetalLBBGPPeer(instance *metallbv1alpha1.BGPPeer) error {
+	objs, err := r.renderObject(instance)
+
+	if err != nil {
+		return fmt.Errorf("Fail to render bgp-peer manifest %v", err)
+	}
+
+	for _, obj := range objs {
+		if err := apply.ApplyObject(context.Background(), r.Client, obj); err != nil {
+			return fmt.Errorf("could not apply (%s) %s/%s err %v", obj.GroupVersionKind(),
+				obj.GetNamespace(), obj.GetName(), err)
+		}
+	}
+
+	return err
+}
+
+func (r *BGPPeerReconciler) syncMetalLBBGPPeers(req ctrl.Request) error {
+	instanceList := &metallbv1alpha1.BGPPeerList{}
+	objs := make([]*unstructured.Unstructured, 0)
+
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      MetalLBConfigMapName,
+			Namespace: req.Namespace,
+		},
+	}
+
+	// Delete the exiting configMap
+	if err := r.Delete(context.Background(), configMap); err != nil {
```

Comment:
This function will only add the BGP peers but will remove any address-pools same for the address-pool package

Reply:
This is a good catch I will see how to fix this